app/models/piece.rb

```diff
@@ -59,6 +50,13 @@ def move_out_of_bounds?(to_row, to_col)
     to_row < 0 || to_row > 7 || to_col < 0 || to_col > 7
   end
 
+  def move_destination_same_side?(to_row, to_col)
+    if game.piece_at(to_row, to_col) && game.piece_at(to_row, to_col).is_black == is_black
```

Comment:
Actually it is a method in game.rb
def piece_at(row, col)
    pieces.where(row: row, col: col).first
  end

I share your concern since the array may contain multiple pieces and there is no guarantee that the first one is the not captured piece. I ensure that the moving piece is not updated in the table until the move checks are complete and a capture occurs.

Let's discuss this today during the VTC - I don't think that piece_at method is safe enough to give us what we want.